src/main/java/com/icthh/xm/tmf/ms/activation/service/SagaServiceImpl.java

```diff
@@ -230,6 +231,26 @@ private boolean isTransactionInCorrectState(SagaEvent sagaEvent, Context context
         return true;
     }
 
+    private boolean isCurrentTaskNotWaitForCondition(SagaEvent sagaEvent, Context context) {
+        SagaTaskSpec taskSpec = context.getTaskSpec();
+        SagaTransaction transaction = context.getTransaction();
+        String txId = context.getTxId();
+
+        try {
+            boolean taskExecutionAllowed = taskExecutor.checkWaitCondition(taskSpec, sagaEvent, transaction);
+            if (!taskExecutionAllowed) {
+                log.info("Task will not executed. Wait condition not happened yet. Transaction id {}.", txId);
+                retryService.retryForTaskWaitCondition(sagaEvent, context.getTaskSpec());
+                return false;
+            }
+        } catch (Throwable t) {
+            log.error("Task will not executed. Error during condition check. Transaction id {}.", txId);
+            retryService.retryForTaskWaitCondition(sagaEvent, context.getTaskSpec());
```

Comment:
better in this case it's retry with state ON_RETRY

Reply:
done